manticore/platforms/linux.py

```diff
@@ -151,6 +151,78 @@ def sync(self):
         """
         return
 
+class ProcSelfMaps(File):
+    def __init__(self, flags, linux):
+        self.filename = "_proc_self_maps_"
+        with open(self.filename, 'w') as mapsFile:
+            print(linux.current.memory.__proc_self__, file = mapsFile)
+        mode = mode_from_flags(flags)
+        if (mode != "rb"):
+            raise EnvironmentError("/proc/self/maps is only supported in read only mode")
+        self.file = open(self.filename, mode)
+
+    def __getstate__(self):
+        state = {"name": self.name, "mode": self.mode, "closed": self.closed}
+        state["pos"] = None if self.closed else self.tell()
+        return state
+
+    def __setstate__(self, state):
+        name = state["name"]
+        mode = state["mode"]
+        closed = state["closed"]
+        pos = state["pos"]
+        try:
+            self.file = open(name, mode)
+            if closed:
+                self.file.close()
+        except IOError:
+            # If the file can't be opened anymore (should not typically happen)
+            self.file = None
+        if pos is not None:
+            self.seek(pos)
+
+    @property
+    def name(self):
+        return self.file.name
+
+    @property
+    def mode(self):
+        return self.file.mode
+
+    @property
+    def closed(self):
+        return self.file.closed
```

Comment:
This. I wonder if `self.file.closed == (self.pos is None) ` 
Dismiss comment if not.

Reply:
I think actually I can remove that and several of the other properties as I used the definition that was already defined in the File object. (I copy pasted properties with the intention of deleting what I did not modify).  Should I look at this property in the file object?